graylog2-web-interface/src/views/components/visualizations/ChartData.js

```diff
@@ -78,9 +86,15 @@ export const chartData = (
   data: Rows,
   chartType: string,
   generator: Generator = _defaultChartGenerator,
-): Array<ChartDefinition> => flow([
-  transformKeys(rowPivots, columnPivots),
-  extractSeries(series.length === 1 ? doNotSuffixTraceForSingleSeries : undefined),
-  removeNulls(),
-  generateChart(chartType, generator),
-])(data);
+  formatSeriesCustom?: ({valuesBySeries: Object, xLabels: Array<any>}) => ExtractedSeries,
```

Comment:
Maybe `customSeriesFormatter` is a better name? We could also define a default value here, removing the need for the conditional assignment below.

Reply:
Good catch with the default value, I change the function name as well.